lib/Dancer2/Core/Role/SessionFactory.pm

```diff
@@ -163,7 +164,7 @@ sub retrieve {
     my %args = ( id => $id, );
 
     $args{data} = $data
-      if $data and ref $data eq 'HASH';
+      if $data and ref $data and reftype $data eq 'HASH';
```

Comment:
True, that. I'm usually the kind of savage who does

```
no warnings 'uninitialized';
... if $foo eq 'bar';
```

instead of 
```
if $foo and $foo eq 'bar';
```

But, yeah, not everybody's cup of tea. :-)